test/contractBridge/executeProposal.js

```diff
@@ -0,0 +1,154 @@
+    });
+
+    it('should create and execute executeProposal successfully', async () => {
+      const proposalSignedData = await Helpers.signDataWithMpc(originDomainID, destinationDomainID, expectedDepositNonce, depositProposalData, resourceID);
+
+        // depositerAddress makes initial deposit of depositAmount
+        assert.isFalse(await BridgeInstance.paused());
+        await TruffleAssert.passes(BridgeInstance.deposit(
+            originDomainID,
+            resourceID,
+            depositData,
+            feeData,
+            { from: depositerAddress }
+        ));
+
+        await TruffleAssert.passes(BridgeInstance.executeProposal(
+            originDomainID,
+            destinationDomainID,
+            expectedDepositNonce,
+            depositProposalData,
+            resourceID,
+            proposalSignedData,
+            { from: relayer1Address }
+        ));
+
+        // check that deposit nonce has been marked as used in bitmap
+        assert.isTrue(await BridgeInstance.isProposalExecuted(originDomainID, expectedDepositNonce));
+
+        // check that tokens are transferred to recipient address
+        const recipientBalance = await ERC20MintableInstance.balanceOf(recipientAddress);
+        assert.strictEqual(recipientBalance.toNumber(), depositAmount);
+    });
+
+    it('executeProposal event should be emitted with expected values', async () => {
+      const proposalSignedData = await Helpers.signDataWithMpc(originDomainID, destinationDomainID, expectedDepositNonce, depositProposalData, resourceID);
+
+        // depositerAddress makes initial deposit of depositAmount
+        assert.isFalse(await BridgeInstance.paused());
+        await TruffleAssert.passes(BridgeInstance.deposit(
+            originDomainID,
+            resourceID,
+            depositData,
+            feeData,
+            { from: depositerAddress }
+        ));
+
+        const proposalTx = await BridgeInstance.executeProposal(
+          originDomainID,
+          destinationDomainID,
+          expectedDepositNonce,
+          depositProposalData,
+          resourceID,
+          proposalSignedData,
+            { from: relayer1Address }
+        );
+
+        TruffleAssert.eventEmitted(proposalTx, 'ProposalExecution', (event) => {
+            return event.originDomainID.toNumber() === originDomainID &&
+                event.destinationDomainID.toNumber() === destinationDomainID &&
+                event.depositNonce.toNumber() === expectedDepositNonce &&
+                event.dataHash === dataHash
+        });
+
+        // check that deposit nonce has been marked as used in bitmap
+        assert.isTrue(await BridgeInstance.isProposalExecuted(originDomainID, expectedDepositNonce));
+
+        // check that tokens are transferred to recipient address
+        const recipientBalance = await ERC20MintableInstance.balanceOf(recipientAddress);
+        assert.strictEqual(recipientBalance.toNumber(), depositAmount);
+    });
+});
```

Comment:
A test for failed executeProposal because of used depositNonce would be also helpful